Network not removed from list after deletion

If a network is deleted while Quasseldroid is running, the network is grayed out to indicate it is disconnected, but is not removed from the list.  Tapping on the network name to view the status buffer results in:

E/AndroidRuntime(11553): java.lang.RuntimeException: Could not dispatch event: class com.iskrembilen.quasseldroid.events.BufferOpenedEvent to handler [EventHandler public void com.iskrembilen.quasseldroid.gui.MainActivity.onBufferOpened(com.iskrembilen.quasseldroid.events.BufferOpenedEvent)]
E/AndroidRuntime(11553):        at com.squareup.otto.Bus.throwRuntimeException(Bus.java:456)
E/AndroidRuntime(11553):        at com.squareup.otto.Bus.dispatch(Bus.java:386)
E/AndroidRuntime(11553):        at com.squareup.otto.Bus.dispatchQueuedEvents(Bus.java:367)
E/AndroidRuntime(11553):        at com.squareup.otto.Bus.post(Bus.java:336)
E/AndroidRuntime(11553):        at com.iskrembilen.quasseldroid.gui.fragments.BufferFragment.openBuffer(BufferFragment.java:391)
E/AndroidRuntime(11553):        at com.iskrembilen.quasseldroid.gui.fragments.BufferFragment.access$000(BufferFragment.java:70)
E/AndroidRuntime(11553):        at com.iskrembilen.quasseldroid.gui.fragments.BufferFragment$BufferListAdapter$3.onClick(BufferFragment.java:598)
E/AndroidRuntime(11553):        at android.view.View.performClick(View.java:4756)
E/AndroidRuntime(11553):        at android.view.View$PerformClick.run(View.java:19749)
E/AndroidRuntime(11553):        at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime(11553):        at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(11553):        at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime(11553):        at android.app.ActivityThread.main(ActivityThread.java:5221)
E/AndroidRuntime(11553):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11553):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(11553):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
E/AndroidRuntime(11553):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
E/AndroidRuntime(11553): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.iskrembilen.quasseldroid.Network.getName()' on a null object reference
E/AndroidRuntime(11553):        at com.iskrembilen.quasseldroid.gui.MainActivity.setTitleAndMenu(MainActivity.java:444)
E/AndroidRuntime(11553):        at com.iskrembilen.quasseldroid.gui.MainActivity.onBufferOpened(MainActivity.java:404)
E/AndroidRuntime(11553):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(11553):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(11553):        at com.squareup.otto.EventHandler.handleEvent(EventHandler.java:89)
E/AndroidRuntime(11553):        at com.squareup.otto.Bus.dispatch(Bus.java:384)
E/AndroidRuntime(11553):        ... 15 more

--- a/QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java
+++ b/QuasselDroid/src/main/java/com/iskrembilen/quasseldroid/gui/MainActivity.java
@@ -395,7 +395,11 @@ public class MainActivity extends ActionBarActivity {
 
     @Subscribe
     public void onBufferOpened(BufferOpenedEvent event) {
-        if (event.bufferId != -1) {
+        NetworkCollection networks = NetworkCollection.getInstance();
+
+        if (event.bufferId != -1
+                && networks.getBufferById(event.bufferId)!=null
+                && networks.getNetworkById(networks.getBufferById(event.bufferId).getInfo().networkId)!=null) {
             openedBuffer = event.bufferId;
             if (event.switchToBuffer) {
                 manager.openDrawer(Gravity.NO_GRAVITY);
@@ -405,7 +409,6 @@ public class MainActivity extends ActionBarActivity {
             }
         }
     }
-
     private void setTitleAndMenu() {
         NetworkCollection networks = NetworkCollection.getInstance();
         Buffer buffer = networks.getBufferById(openedBuffer);

